### ISSUE TITLE
Trouble restoring box_vectors for openMM

### ISSUE THREAD

**Issue by hejung** [openpathsampling/openpathsampling#789]
I just updated openMM to 7.2.2 and now have trouble when loading from file. It does not matter if it is a just created file or a file that was created with the old version.

The problem seems to be that when loading a snapshot from storage its box_vectors are now set to a masked array instead of an array. This happens for new and old storage files and lets fail the check if box_vectors are None in Trajectory.to_mdtraj()
I guess the problem is somewhere in the reconstruction of the snapshots when loading them from storage.

minimal example:
```python
# engine setup etc.
forcefield = app.ForceField('amber99sbildn.xml', 'tip3p.xml')
pdb = app.PDBFile("AD_initial_frame.pdb")
system = forcefield.createSystem(
    pdb.topology, 
    nonbondedMethod=app.PME, 
    nonbondedCutoff=1.0*unit.nanometers,
    constraints=app.HBonds, 
    rigidWater=True,
    ewaldErrorTolerance=0.0005
)
integrator = VVVRIntegrator(
    600*unit.kelvin, 
    1.0/unit.picoseconds, 
    2.0*unit.femtoseconds)
integrator.setConstraintTolerance(0.00001)

engine_options = {
    'n_frames_max': 2000,
    'nsteps_per_frame': 10,
}

template = peng_omm.snapshot_from_pdb("../initial_TPs/AD_initial_frame.pdb")
engine = peng_omm.Engine(template.topology, 
    system, 
    integrator, 
    options=engine_options
)
engine.name = '600K'
engine.initialize(platform=None) # platform='CPU'

engine.current_snapshot = template
engine.minimize()

# the interesting part:
storage = paths.Storage(filename='test.nc', mode='w', template=template)

trial_traj = engine.generate_n_frames(100)

print(repr(trial_traj[0].box_vectors))
# returns : 
# Quantity(value=array([[2.6063, 0.    , 0.    ],
#       [0.    , 2.6063, 0.    ],
#      [0.    , 0.    , 2.6063]], dtype=float32), unit=nanometer)

storage.save(trial_traj)
storage.sync_all()

new_store1 =  paths.Storage(filename='test.nc', mode='r')
# if I now do a load it still works, i.e. before closing the old storage
load1 = new_store1.trajectories[-1]

print(repr(load1[0].box_vectors)))
# returns : 
# Quantity(value=array([[2.6063, 0.    , 0.    ],
#       [0.    , 2.6063, 0.    ],
#      [0.    , 0.    , 2.6063]], dtype=float32), unit=nanometer)

storage.sync_all()
storage.close()
new_store1.close()

new_store2 =  paths.Storage(filename='test.nc', mode='r')
# if I now do a load it still works, i.e. before restarting the python kernel
load2 = new_store2.trajectories[-1]

print(repr(load2[0].box_vectors))
# returns : 
# Quantity(value=array([[2.6063, 0.    , 0.    ],
#       [0.    , 2.6063, 0.    ],
#      [0.    , 0.    , 2.6063]], dtype=float32), unit=nanometer)

# after restarting the ipython notebook
new_store3 =  paths.Storage(filename='test.nc', mode='r')
# if I now do a load it still works, i.e. before restarting the python kernel
load3 = new_store3.trajectories[-1]

print(repr(load3[0].box_vectors))
# returns : 
# Quantity(value=masked_array(
#  data=[[2.606300115585327, 0.0, 0.0],
#        [0.0, 2.606300115585327, 0.0],
#        [0.0, 0.0, 2.606300115585327]],
#  mask=[[False, False, False],
#        [False, False, False],
#        [False, False, False]],
#  fill_value=9.96921e+36,
#  dtype=float32), unit=nanometer)'
```

A quick fix is replacing [trajectory.py line 567](https://github.com/dwhswenson/openpathsampling/blob/770ce4f188cc7612dae07fc3bb834c9152181719/openpathsampling/engines/trajectory.py#L567)
with 
```python
if not box_vectors is None:
            if not np.any([np.array(v) for v in box_vectors]):
```
however I have no idea if this is a good idea?

**Comment by dwhswenson** [comment]
Can you double-check that you're fully up to date? This sounds very much like one of the problems that was fixed in #776. It may have been an update to NetCDF that caused the problem (see that issue), as opposed to OpenMM. Last I checked `simtk.unit` has problems with masked arrays, which was what #776 fixed.

**Comment by hejung** [comment]
You are correct. I did not pull the latest changes into my pathsamp_hook for a long time since it was so tedious with the different folder layout when I first started working on it.
But this fixes it and lets me work with the newest and shiniest stuff OPS has to offer :)
Also I just realized I could have done that ~6 month ago when the new folder layout made its way into master...
